libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoslSyncDownTarget.java

```diff
@@ -91,11 +91,16 @@ public JSONArray continueFetch(SyncManager syncManager) throws IOException, JSON
         return null;
     }
 
+    @Override
+    public Set<String> getListOfRemoteIds(SyncManager syncManager, Set<String> localIds) {
+        // TODO: Do a SOQL query for each object type with 'IN' clause from localIds.
```

Comment:
TODO: ```SOSL``` implementation. This is not going to be fun - need to iterate and build separate ```SOQL``` queries for different entities.